### ISSUE TITLE
Documentation for AdamW is wrong

### ISSUE THREAD

**Issue by thomasahle** [pytorch/pytorch#68482]
## 🐛 Bug

The second line of the formula for `AdamW` here: https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html is wrong

## To Reproduce

Read the line `theta_t = theta_t  - gamma lambda theta_{t-1}`.

## Expected behavior

This should say `g_t = g_t - lambda theta_{t-1}`

It is clear that updating `theta_t` at this step doesn't make sense, since it isn't read again in the code.

Changing the update to `g_t` makes the algorithm match Adam https://pytorch.org/docs/stable/generated/torch.optim.Adam.html (with lambda != 0).

The paper referred to in the documentation: https://arxiv.org/pdf/1711.05101.pdf
Does it like this
`g_t = g_t + lambda theta_{t-1}`,
but the difference of +/- on lambda probably doesn't matter much.

cc @brianjo @mruberry @vincentqb @jbschlosser @albanD

**Comment by albanD** [comment]
There is indeed a typo in the last lines. They should be using `theta_t` in the last two lines.
But the line you linked is correct, it is performing the weight decay before the adamw step itself.

**Comment by thomasahle** [comment]
I find it a little odd to change theta (the parameters), but maybe it's equivalent.
The paper doesn't do it like that (in Algorithm 2), but I agree that this is what the pytorch code seems to be doing.
